### ISSUE TITLE
Understanding the Reinforcement Learning (RL) Framework

### ISSUE THREAD

**Issue by phanindra-max** [phanindra-max/SwapStack#2]
- [ ] Understanding Markov Decision Process (MDP) assumptions and limitations: markov property, stationarity, limitations.
- [ ] Understanding sequential decision making: actions affect future states and rewards.
- [ ]  Understanding terminal states and episodic tasks: pseudo-labeling epidode ends when all unlabeled data is processed.
- [ ] State space design: selecting features of unlabeled examples, model predictions, and uncertainty measures 
        (e.g. [feature1, ..., featuren, softmax_predictions, cross_entropy]). 
- [ ] Action space design: assign a pseudo-label or skip labeling an example (e.g. {0, 1, ..., 9} for MNIST).
- [ ] Reward design: reward high-confidence correct pseudo-labels and penalize incorrect assignments.
- [ ] Algorithm selection: evaluate classical RL methods such as Q-learning, and deep RL methods such as Policy Gradient approaches.
